compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -645,17 +653,42 @@ fn print_crate_info(
     } else {
         None
     };
+
+    let mut output_io: Box<dyn Write> = if let Some(output_file_path) = &sess.io.output_file {
+        match File::create(output_file_path) {
+            Ok(output_file) => Box::new(output_file),
+            Err(err) => early_error(
+                sess.opts.error_format,
+                &format!("failed to create {}: {}", output_file_path.display(), err),
+            ),
+        }
+    } else {
+        Box::new(io::stdout())
+    };
+
+    fn write_output(output_io: &mut dyn Write, args: fmt::Arguments<'_>) {
+        if let Err(_) = output_io.write_fmt(args) {
+            rustc_errors::FatalError.raise();
+        }
+    }
```

Comment:
Can this use `pretty::write_or_print`, or adapt that function such that it can be used?

Reply:
Yes, done.